### PR TITLE
Feature/3114/create laravel console command for new subscriber

### PR DESCRIPTION
**Pull request by rajatsajwan**
Target #3114

Description:
multiple   jobIds  can be their

### CODE REVIEW

File: app/Console/Commands/sendNewSubscriber.php [hunk 1]
```diff
@@ -54,15 +54,17 @@ public function handle()
...
-            $jobId = Application::where('hr_applicant_id', $applicant->id)->value('hr_job_id');
-            if ($jobId) {
-                $subscriptionLists = Job::where('id', $jobId)->value('title');
-                if ($subscriptionLists) {
-                    try {
-                        $applicationService = new ApplicationService();
-                        $applicationService->addSubscriberToCampaigns($data, $subscriptionLists);
-                    } catch (\Exception $e) {
-                        $this->error('Error occurred while sending data to Campaign');
+            $jobIds = Application::where('hr_applicant_id', $applicant->id)->pluck('hr_job_id');
```
Comment (mohitgusain) [inline app/Console/Commands/sendNewSubscriber.php:57]: What is `jobIds`? is it a collection or array ?

File: app/Console/Commands/sendNewSubscriber.php [hunk 2]
```diff
@@ -54,15 +54,17 @@ public function handle()
...
+            foreach ($jobIds as $jobId) {
```
Comment (mohitgusain) [inline app/Console/Commands/sendNewSubscriber.php:58]: Can you add a check here? if $jobIds is not null then foreach loop will execute.

File: app/Console/Commands/sendNewSubscriber.php [hunk 3]
```diff
@@ -54,15 +54,17 @@ public function handle()
...
+                if ($jobId) {
```
Comment (mohitgusain) [inline app/Console/Commands/sendNewSubscriber.php:59]: This check will not be required then

**Comment by mohitgusain** [comment]
@rajatsajwan make sure that you have tested this code in you locally

**Comment by rajatsajwan** [comment]
I am testing this code in my local.

**Review by mohitgusain — APPROVED** [review]

File: app/Console/Commands/sendNewSubscriber.php [hunk 4]
```diff
@@ -54,10 +54,17 @@ public function handle()
...
-                if ($subscriptionLists) {
+            $jobIds = Application::where('hr_applicant_id', $applicant->id)->pluck('hr_job_id')->toArray();
+            if (! empty($jobIds)) {
+                $uniqueJobIds = array_unique($jobIds);
+                $subscriptionLists = [];
+
+                foreach ($uniqueJobIds as $jobId) {
+                    $subscriptionList = Job::where('id', $jobId)->value('title');
+                    $subscriptionLists[] = $subscriptionList;
+                }
```
Comment (rathorevaibhav) [inline app/Console/Commands/sendNewSubscriber.php:65]: there are multiple queries happening here. Can't we run one query which returns titles for all the job ids?

File: Modules/HR/Http/Controllers/Recruitment/ApplicantController.php [hunk 5]
```diff
@@ -56,12 +56,6 @@ public function store(ApplicantRequest $request)
         $validated = $request->validated();
         $job_title = Job::where('opportunity_id', $validated['opportunity_id'])->first();
 
-        // try {
```
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply Modules/HR/Http/Controllers/Recruitment/ApplicantController.php:59]: why is this code removed?

**Review by rathorevaibhav — CHANGES_REQUESTED** [review]

**Comment by rajatsajwan** [comment]
// try {

**Review by rathorevaibhav — APPROVED** [review]

**Pull request merged**